geoh5py/shared/conversion/grid2d_to_geoimage.py

```diff
@@ -0,0 +1,172 @@
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+from PIL import Image
+
+from ... import objects
+from .base_conversion import ConversionBase
+
+if TYPE_CHECKING:
+    from ...objects import GeoImage, Grid2D
+
+
+class Grid2dToGeoImage(ConversionBase):
+    """
+    Convert a :obj:geoh5py.objects.grid2d.Grid2D object
+    to a georeferenced :obj:geoh5py.objects.geo_image.GeoImage object.
+    """
+
+    def __init__(self, entity: Grid2D):
+        """
+        :param entity: the :obj:geoh5py.objects.grid2d.Grid2D to convert.
+        """
+        if not isinstance(entity, objects.Grid2D):
+            raise TypeError(f"Entity must be 'Grid2D', {type(entity)} passed instead")
+
+        super().__init__(entity)
+        self.entity: Grid2D
+
+    def grid_to_tag(self) -> dict:
+        """
+        Compute the tag dictionary of the Grid2D as required by the
+        :obj:geoh5py.objects.geo_image.GeoImage object.
+        :return: a dictionary of the tag.
+        """
+        if not isinstance(self.entity.v_cell_size, np.ndarray) or not isinstance(
+            self.entity.u_cell_size, np.ndarray
+        ):
+            raise AttributeError("The Grid2D has no geographic information")
+
+        if self.entity.u_count is None or self.entity.v_count is None:
+            raise AttributeError("The Grid2D has no number of cells")
+
+        u_origin, v_origin, z_origin = self.entity.origin.item()
+        v_oposite = v_origin + self.entity.v_cell_size * self.entity.v_count
+        tag = {
+            256: (self.entity.u_count,),
+            257: (self.entity.v_count,),
+            33550: (
+                self.entity.u_cell_size[0],
+                self.entity.v_cell_size[0],
+                0.0,
+            ),
+            33922: (
+                0.0,
+                0.0,
+                0.0,
+                u_origin,
+                v_oposite,
+                z_origin,
+            ),
+        }
+
+        return tag
+
+    def data_to_pil_format(self, data_name: str) -> np.array:
+        """
+        Convert a data of the Grid2D to a numpy array
+        with a format compatible with :obj:`PIL.Image` object.
+        :param data_name: the name of the data to extract.
+        :return: the data formatted with the right shape,
+        beetweem 0 and 255, as uint8.
+        """
+        # get the data
+        data = self.entity.get_data(data_name)[0].values
+
+        # reshape them
+        data = data.reshape(self.entity.v_count, self.entity.u_count)[::-1]
+
+        # normalize them #todo: change the no number value in np.nan
+        min_, max_ = np.nanmin(data), np.nanmax(data)
+        data = (data - min_) / (max_ - min_)
+        data *= 255
+
+        return data.astype(np.uint8)
+
+    def __call__(self, data_list: list | str, **geoimage_kwargs) -> GeoImage:
```

Comment:
Can we implement to accept a `Data` or `UUID` as well? String names are too dangerous to be the only thing we rely on...